### PR TITLE
Feature/ens

### PR DESCRIPTION
**Pull request by JuliusBrain**
Important changes:
- [x] Tests are rewritten to use chai and be more like BDD tests

Mandatory:
- [x] Added documentation.

Closes #

### CODE REVIEW

File: client/contracts/truffle/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+## ENS Explained
+ENS is the Ethereum Name Service, a distributed, open, and extensible naming system based on the Ethereum blockchain.
```
Comment (rgeraldes) [inline client/contracts/truffle/README.md]: Replace Ethereum references with Kowala.

File: client/contracts/truffle/contracts/ens/ENS.sol [hunk 2]
```diff
@@ -0,0 +1,26 @@
+pragma solidity ^0.4.18;
```
Comment (rgeraldes) [inline client/contracts/truffle/contracts/ens/ENS.sol]: pragma solidity 0.4.24;

File: client/contracts/truffle/contracts/ens/ENSRegistry.sol [hunk 3]
```diff
@@ -0,0 +1,99 @@
+pragma solidity ^0.4.18;
```
Comment (rgeraldes) [inline client/contracts/truffle/contracts/ens/ENSRegistry.sol]: pragma solidity 0.4.24;

File: client/contracts/truffle/contracts/ens/ENSRegistry.sol [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    struct Record {
+        address owner;
+        address resolver;
+        uint64 ttl;
+    }
+
+    mapping (bytes32 => Record) records;
+
+    // Permits modifications only by the owner of the specified node.
+    modifier only_owner(bytes32 node) {
```
Comment (rgeraldes) [inline client/contracts/truffle/contracts/ens/ENSRegistry.sol]: only_owner vs onlyOwner

File: client/contracts/truffle/contracts/ens/FIFSRegistrar.sol [hunk 5]
```diff
@@ -0,0 +1,36 @@
+pragma solidity ^0.4.18;
```
Comment (rgeraldes) [inline client/contracts/truffle/contracts/ens/FIFSRegistrar.sol]: pragma solidity 0.4.24;

File: client/contracts/truffle/contracts/ens/FIFSRegistrar.sol [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import "./ENS.sol";
+
+/**
+ * A registrar that allocates subdomains to the first person to claim them.
+ */
+contract FIFSRegistrar {
+    ENS ens;
+    bytes32 rootNode;
+
+    modifier only_owner(bytes32 subnode) {
```
Comment (rgeraldes) [inline client/contracts/truffle/contracts/ens/FIFSRegistrar.sol]: only_owner vs onlyOwner

File: client/contracts/truffle/contracts/ens/PublicResolver.sol [hunk 7]
```diff
@@ -0,0 +1,238 @@
+pragma solidity ^0.4.18;
```
Comment (rgeraldes) [inline client/contracts/truffle/contracts/ens/PublicResolver.sol]: pragma solidity 0.4.24;

File: client/contracts/truffle/test/ens/ENS.test.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
+/* global artifacts, contract, it, beforeEach, web3 */
+/* eslint no-unused-expressions: 0 */
+/* eslint consistent-return: 0 */
+/* eslint-disable max-len */
+
+const ENS = artifacts.require('ENSRegistry.sol');
+
```
Comment (rgeraldes) [inline client/contracts/truffle/test/ens/ENS.test.js]: remove extra space

File: client/contracts/truffle/migrations/2_deploy_contracts.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
+/* global artifacts, web3 */
+/* eslint-disable max-len */
+
+const ENS = artifacts.require('./ens/ENSRegistry.sol');
+const FIFSRegistrar = artifacts.require('./ens/FIFSRegistrar.sol');
+const PublicResolver = artifacts.require('./ens/PublicResolver.sol');
+
```
Comment (rgeraldes) [inline client/contracts/truffle/migrations/2_deploy_contracts.js]: remove extra space

File: client/contracts/truffle/README.md (hunk 1 above)
Comment (JuliusBrain) [reply client/contracts/truffle/README.md]: Is that necessary? We are using ENS and if someone will check the repo later and will see an ENS name, they will know what is this for

Comment (rgeraldes) [reply client/contracts/truffle/README.md]: We should not use ENS in the first place - NS should be enough.

File: client/contracts/truffle/README.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
+## NS Explained
+NS is the Ethereum Name Service, a distributed, open, and extensible naming system based on the Ethereum blockchain.
```
Comment (rgeraldes) [inline client/contracts/truffle/README.md]: Kowala

File: client/contracts/truffle/README.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+NS can be used to resolve Ethereum addresses, from human-readable form like “kowala.tech” into machine-readable identifiers, including Ethereum addresses.
```
Comment (rgeraldes) [inline client/contracts/truffle/README.md]: replace Ethereum with Kowala

**Comment by rgeraldes** [comment]
covered in #561

**Pull request closed without merging**